pkg/remotewrite/prometheus.go

```diff
@@ -23,121 +29,109 @@ func MapTagSet(t *metrics.TagSet) []*prompb.Label {
 	return labels
 }
 
-func MapSeries(ts metrics.TimeSeries) prompb.TimeSeries {
-	return prompb.TimeSeries{
-		Labels: append(MapTagSet(ts.Tags), &prompb.Label{
-			Name:  "__name__",
-			Value: fmt.Sprintf("%s%s", defaultMetricPrefix, ts.Metric.Name),
-		}),
-	}
+// MapSeries converts a k6 time series into
+// the equivalent set of Labels (name+tags) as expected from the
+// Prometheus' data model.
+//
+// The labels are lexicographic sorted as required
+// from the Remote write's specification.
+func MapSeries(series metrics.TimeSeries) []*prompb.Label {
+	lbls := append(MapTagSet(series.Tags), &prompb.Label{
+		Name:  namelbl,
+		Value: fmt.Sprintf("%s%s", defaultMetricPrefix, series.Metric.Name),
+	})
+	sort.Slice(lbls, func(i int, j int) bool {
+		return lbls[i].Name < lbls[j].Name
+	})
+	return lbls
 }
 
-func MapTrend(series metrics.TimeSeries, t time.Time, sink *trendSink) []*prompb.TimeSeries {
+// MapTrendAsGauges converts a k6 time series and its relative
+// Sink into the equivalent TimeSeries model as defined from
+// the Remote write specification.
+func MapTrendAsGauges(series metrics.TimeSeries, t time.Time, sink *metrics.TrendSink) []*prompb.TimeSeries {
```

Comment:
What are our plans for the neighbor PR #57?

I'm asking because if we want to continue to work on it, it make sense to implement https://github.com/grafana/xk6-output-prometheus-remote/pull/57/files#r1025272534

Reply:
I will try to do it here, I added a comment directly in #57. I think there is the risk to expand too much the scope in this PR. If I will add too many changes then I will submit a dedicated PR on top of this. @olegbespalov WDYT?

Reply:
That's why I'm confused a bit. I guess these two PRs are changing the same parts of the logic/structs, which will lead to conflict. Maybe we can combine them into one with different commits?

Reply:
Sorry, I thought it was obvious but it was a bad assumption. I'm splitting from #57 all the things not strictly related to Histogram. So we can focus there mostly on reviewing that part.

Reply:
:+1: